examples/tinytest/src/main.rs

```diff
@@ -2,66 +2,89 @@ extern crate tinyfiledialogs;
 extern crate libc;
 
 use std::ffi::{CString, CStr};
-use libc::c_int;
+use libc::{c_int, c_char, c_uchar};
 
-fn message_box(title: &CStr, message: &CStr, dialog_type: &CStr, icon: &CStr, button: c_int) {
+fn message_box(title: &CStr, message: &CStr,
+               dialog_type: &CStr,
+               icon: &CStr, button: c_int) -> c_int {
     unsafe {
-        tinyfiledialogs::tinyfd_messageBox(
+        let result: c_int = tinyfiledialogs::tinyfd_messageBox(
             title.as_ptr(),
             message.as_ptr(),
             dialog_type.as_ptr(),
             icon.as_ptr(),
             button);
+
+        result
     }
 }
 
-fn input_box(title: &CStr, message: &CStr, default: &CStr) {
+fn input_box(title: &CStr,
+             message: &CStr,
+             default: &CStr) -> *const c_char {
     unsafe {
-        tinyfiledialogs::tinyfd_inputBox(title.as_ptr(), message.as_ptr(), default.as_ptr());
+        let result: *const c_char = tinyfiledialogs::tinyfd_inputBox(title.as_ptr(), message.as_ptr(), default.as_ptr());
+        
+        result
     }
 }
 
-fn save_file_dialog(title: &CStr, path: &CStr, num_patterns: c_int, filter_patterns: &CStr, des: &CStr) {
+fn save_file_dialog(title: &CStr, path: &CStr,
+                    num_patterns: c_int,
+                    filter_pattern_one: &CStr, des: &CStr) -> *const c_char {
+    let filter_patterns : [&CStr; 1] = [filter_pattern_one];
     unsafe {
-        tinyfiledialogs::tinyfd_saveFileDialog(
+        let result: *const c_char = tinyfiledialogs::tinyfd_saveFileDialog(
             title.as_ptr(),
             path.as_ptr(),
             num_patterns,
-            filter_patterns.as_ptr(),
+            filter_patterns.iter().map(|c| c.as_ptr()).collect::<Vec<*const c_char>>().as_ptr(),
```

Comment:
This needs to be in a local variable in order to be safe; otherwise the pointer is pointing at memory that is freed as soon as the collect() expression goes out of scope.